js/src/client.ts

```diff
@@ -705,7 +736,14 @@ export class Client {
       data.trace_id !== undefined &&
       data.dotted_order !== undefined
     ) {
-      this.appendRunCreateToAutoBatchQueue({ action: "update", item: data });
+      if (run.end_time !== undefined && data.parent_run_id === undefined) {
+        // Trigger a batch as soon as a root trace ends and block to ensure trace finishes
+        // in serverless environments.
+        await this.processRunOperation({ action: "update", item: data }, true);
+        return;
+      } else {
+        void this.processRunOperation({ action: "update", item: data });
```

Comment:
Cool so this one is non blocking then?

Reply:
Yes, we only want to block on root run ends